objects/client.c

```diff
@@ -797,103 +936,168 @@
- * @tfield[opt] boolean table.decorations.title
- * @tfield[opt] boolean table.decorations.menu
- * @tfield[opt] boolean table.decorations.minimize
- * @tfield[opt] boolean table.decorations.maximize
- * @tfield[opt] string table.input_mode
- * @tfield[opt] table table.status
- * @tfield[opt] boolean table.status.tearoff_window
+ * @tparam table hints The hints.
+ * @tparam[opt] boolean hints.functions.all
+ * @tparam[opt] boolean hints.functions.resize
+ * @tparam[opt] boolean hints.functions.move
+ * @tparam[opt] boolean hints.functions.minimize
+ * @tparam[opt] boolean hints.functions.maximize
+ * @tparam[opt] boolean hints.functions.close
+ * @tparam[opt] boolean hints.decorations.all
+ * @tparam[opt] boolean hints.decorations.border
+ * @tparam[opt] boolean hints.decorations.resizeh
+ * @tparam[opt] boolean hints.decorations.title
+ * @tparam[opt] boolean hints.decorations.menu
+ * @tparam[opt] boolean hints.decorations.minimize
+ * @tparam[opt] boolean hints.decorations.maximize
+ * @tparam[opt] string hints.input_mode This is either `modeless`,
+ *  `primary_application_modal`, `system_modal`,
+ *  `full_application_modal` or `unknown`.
+ * @tparam[opt] boolean hints.status.tearoff_window
  * @propemits false false
  */
 
 /**
  * Set the client sticky, i.e. available on all tags.
  *
+ * Please note that AwesomeWM implements `sticky` clients
+ * per screens rather than globally like some other
+ * implementations.
+ *
  * @property sticky
- * @param boolean
+ * @tparam boolean sticky
  * @propemits false false
+ * @see skip_taskbar
  */
 
 /**
  * Indicate if the client is modal.
  *
+ * A transient window is a client that "belongs" to another
+ * client. If the client is also `modal`, then it always has
+ * to be on top of the other window *and* the parent client
+ * cannot be focused while the child client exists.
+ * This is common for "Save as" dialogs or other dialogs where
+ * isn't possible to modify the content of the "parent" client
+ * while the dialog is open.
+ *
+ * However, `modal` isn't a requirement for using the `transient_for`
+ * concept. "Tools" such as popup palette in canvas-and-palettes
+ * applications can belong to each other without being modal.
+ *
  * @property modal
- * @param boolean
+ * @tparam boolean modal
  * @propemits false false
+ * @see transient_for
  */
 
 /**
  * True if the client can receive the input focus.
  *
+ * The client wont get focused even when the user
+ * click on it.
+ *
  * @property focusable
- * @param boolean
+ * @tparam boolean focusable
  * @propemits false false
+ * @see shape_input
+ * @see client.focus
+ * @see active
+ * @see activate
  */
 
 /**
  * The client's bounding shape as set by awesome as a (native) cairo surface.
  *
- * @see gears.surface.apply_shape_bounding
+ * The bounding shape is the outer shape of the client. It is outside of the
+ * border.
+ *
+ * Do not use this directly unless you want total control over the shape (such
```

Comment:
Can one even do so? I think `awful.client.shape` will interfere as soon as someone starts e.g. `xeyes` and now two implementations will "fight" over the shape.

Reply:
I also wondered about this when writing this. Should we deprecate this and officially call it a private API?

Reply:
Dunno. One could "get rid" of `awful.client.shape` via e.g. `package.loaded["awful.client.shape"] = {}` and write an own version in the config.

Reply:
Ok. That seems hacky. I will think about it before we get to 5.0 to see if hiding those APIs is the best way forward. We will need to come up with some kind of plans for shape with holes in them. Maybe compositing is the way to go. I can't really think of good use cases where it would be worth it, but maybe I lack imagination.